etna/datasets/utils.py

```diff
@@ -91,13 +91,10 @@ def duplicate_data(df: pd.DataFrame, segments: Sequence[str], format: str = Data
         raise ValueError("There should be 'timestamp' column")
 
     # construct long version
-    segments_results = []
-    for segment in segments:
-        df_segment = df.copy()
-        df_segment["segment"] = segment
-        segments_results.append(df_segment)
-
-    df_long = pd.concat(segments_results, ignore_index=True)
+    n_segments, n_timestamps = len(segments), df.shape[0]
+    df_long = pd.DataFrame(np.tile(A=df, reps=(n_segments, 1)), columns=df.columns)
```

Comment:
Are you sure it works fine in all cases? For example, with categorical dtype, "Int64" dtype (int with NaNs) etc.

Reply:
What about solution with `.iloc` like [here](https://stackoverflow.com/a/47149790)?

Reply:
I hoped that we had tests for these cases)

Reply:
You could enrich the tests with these cases